### ISSUE TITLE
Does DQD assume that the connections details must be stored in a variable called "connectionDetails"?

### ISSUE THREAD

**Issue by yampeku** [OHDSI/DataQualityDashboard#232]
I will have 2 different CDMs in my machine, so I preemptively created a connection details variable with a name different to "connectionDetails" and passed it to the process. Every test fails and I'm seeing lots of '<NA>' and things like

```
1.1  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       1      1
1.2  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       2      1
1.3  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       3      1
1.4  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       4      1
1.5  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       5      1
1.6  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       6      1
1.7  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       7      1
1.8  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       8      1
1.9  Completeness               Validation      NA objeto 'connectionDetails' no encontrado       9      1
1.10 Completeness               Validation      NA objeto 'connectionDetails' no encontrado      10      1
1.11 Completeness               Validation      NA objeto 'connectionDetails' no encontrado      11      1
1.12 Completeness               Validation      NA objeto 'connectionDetails' no encontrado      12      1
1.13 Completeness               Validation      NA objeto 'connectionDetails' no encontrado      13      1
1.14 Completeness               Validation      NA objeto 'connectionDetails' no encontrado      14      1
1.15 Completeness               Validation      NA objeto 'connectionDetails' no encontrado      15      1
2.1   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      16      1
2.2   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      17      1
2.3   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      18      1
2.4   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      19      1
2.5   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      20      1
2.6   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      21      1
2.7   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      22      1
2.8   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      23      1
2.9   Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      24      1
2.10  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      25      1
2.11  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      26      1
2.12  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      27      1
2.13  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      28      1
2.14  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      29      1
2.15  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      30      1
2.16  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      31      1
2.17  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      32      1
2.18  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      33      1
2.19  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      34      1
2.20  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      35      1
2.21  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      36      1
2.22  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      37      1
2.23  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      38      1
2.24  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      39      1
2.25  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      40      1
2.26  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      41      1
2.27  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      42      1
2.28  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      43      1
2.29  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      44      1
2.30  Conformance  Relational Verification      NA objeto 'connectionDetails' no encontrado      45      1
```

(error in English would be "object 'connectionDetails' not found")
I'll launch analysis again by creating this object, but seems something that could cause errors to unsuspecting folks

**Comment by yampeku** [comment]
If I create a variable called 'connectionDetails' equal to my connection then process works perfectly

**Comment by actualben** [comment]
This seems to be similar to issue #142 (closed but not resolved), and a problem in #269. 

The problem seems to be that in `execution.R` in the following function:

```R
.needsAutoCommit <- function(connection) {
  autoCommit <- FALSE
  if (!is.null(connection)) {
    if (inherits(connection, "DatabaseConnectorJdbcConnection")) {
      if (connectionDetails$dbms %in% c("postgresql", "redshift")) {
        autoCommit <- TRUE
      }
    }
  }
  autoCommit
}
```

On the line: https://github.com/OHDSI/DataQualityDashboard/blob/fcbc88ebb8006ffcdcc42e289a034e559c0a16e5/R/execution.R#L743

is `connectionDetails` is coming from the global scope.

**There is an existing PR #270 which attempts to fix this.**
